### PR TITLE
formatted docstring of sympy.geometry.line.py module according to the documemtation

### PR DESCRIPTION
**Pull request by viratkohli1969**
#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/geometry/line.py [hunk 1]
```diff
@@ -28,7 +28,6 @@
 from sympy.functions.elementary.piecewise import Piecewise
 from sympy.functions.elementary.trigonometric import (_pi_coeff as pi_coeff, acos, tan, atan2)
 from sympy.geometry.exceptions import GeometryError
-from sympy.geometry.util import intersection
```
Comment (czgdp1807) [inline sympy/geometry/line.py:31]: Why removed? Quality tests automatically detect unused imports if they are present.

File: sympy/geometry/line.py [hunk 2]
```diff
@@ -358,12 +398,14 @@ def direction(self):
         Returns
         =======
 
-        p : a Point; the ray from the origin to this point is the
-            direction of `self`
+        Point :
+            the ray from the origin to this point is the
+            direction of `self`.
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:403]: There can be a better description than this. I don't think "the direction of `self`" is any useful

File: sympy/geometry/line.py [hunk 3]
```diff
@@ -385,26 +435,23 @@ def direction(self):
         return self.p2 - self.p1
 
     def intersection(self, other):
-        """The intersection with another geometrical entity.
+        """Generate the intersection with another geometrical entity.
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:438]: I don't think Generate is the right word here. Also, please move this to the next line.

File: sympy/geometry/line.py [hunk 4]
```diff
@@ -550,20 +604,23 @@ def is_parallel(l1, l2):
...
-        False : otherwise.
+        TypeError
+            if `l1` and `l2` both are not Linear entities.
 
-        See Also
-        ========
+        Returns
+        =======
 
-        coefficients
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:562]: Please add this See Also section back after Examples

File: sympy/geometry/line.py [hunk 5]
```diff
@@ -623,12 +687,14 @@ def is_perpendicular(l1, l2):
...
         False
+
+        perpendicularity over 3D entities.
         >>> from sympy import Point3D, Line3D
         >>> p1, p2, p3 = Point3D(0, 0, 0), Point3D(1, 1, 1), Point3D(-1, 2, 0)
         >>> l1, l2 = Line3D(p1, p2), Line3D(p2, p3)
         >>> l1.is_perpendicular(l2)
         False
-        >>> p4 = Point3D(5, 3, 7)
+        >>> p4 = Point3D(1, 1, -1)
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:697]: Any reason for changing this example?

File: sympy/geometry/line.py [hunk 6]
```diff
@@ -640,26 +706,48 @@ def is_perpendicular(l1, l2):
         return S.Zero.equals(l1.direction.dot(l2.direction))
 
     def is_similar(self, other):
-        """
-        Return True if self and other are contained in the same line.
+        """Whether lines are same or different.
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:709]:
```suggestion
        """
        Checks whether lines are similar or different.
```

File: sympy/geometry/line.py [hunk 7]
```diff
@@ -640,26 +706,48 @@ def is_perpendicular(l1, l2):
...
+
+        Parameters
+        ==========
+
+        other : GeometryEntity
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:714]:
```suggestion
        other : GeometryEntity
            The line whose similarity is to be compared.
```

File: sympy/geometry/line.py [hunk 8]
```diff
@@ -640,26 +706,48 @@ def is_perpendicular(l1, l2):
...
+
+        Returns
+        =======
+
+        bool :
+            True : if self and other are contained in the same line.
+            False : otherwise.
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:721]: Don't think this is much useful to the user. 
These messages should be regarding the lines, not the implimentation details.

File: sympy/geometry/line.py [hunk 9]
```diff
@@ -726,33 +815,36 @@ def parallel_line(self, p):
         Returns
         =======
 
-        line : Line
-
-        See Also
-        ========
-
-        is_parallel
+        Line:
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:818]:
```suggestion
        line : Line
```

File: sympy/geometry/line.py [hunk 10]
```diff
@@ -769,33 +861,36 @@ def perpendicular_line(self, p):
         Returns
         =======
 
-        line : Line
-
-        See Also
-        ========
-
-        sympy.geometry.line.LinearEntity.is_perpendicular, perpendicular_segment
+        Line :
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:864]:
```suggestion
        line : Line
```

File: sympy/geometry/line.py (hunk 9 above)
Comment (Sc0rpi0n101) [reply sympy/geometry/line.py:818]: It should be 
```
entity : Type
    Description
```

File: sympy/geometry/line.py [hunk 11]
```diff
@@ -817,21 +912,19 @@ def perpendicular_segment(self, p):
         Returns
         =======
 
-        segment : Segment
+        Segment :
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:915]:
```suggestion
        segment : Segment
```

File: sympy/geometry/line.py [hunk 12]
```diff
@@ -870,22 +970,31 @@ def points(self):
         Returns
         =======
 
-        points : tuple of Points
-
-        See Also
-        ========
-
-        sympy.geometry.point.Point
+        Point :
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:973]:
```suggestion
        points : Tuple of Points
```

File: sympy/geometry/line.py [hunk 13]
```diff
@@ -897,18 +1006,18 @@ def projection(self, other):
...
         Raises
         ======
 
         GeometryError
             When method is unable to perform projection.
 
+        Returns
+        =======
+
+        Point or LinearEntity (Line, Ray, Segment) :
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1018]:
```suggestion
        projection : Point or LinearEntity (Line, Ray, Segment)
```

File: sympy/geometry/line.py [hunk 14]
```diff
@@ -897,18 +1006,18 @@ def projection(self, other):
...
+            Projection of ``other`` on this linear entity.
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1019]: The message should be about what other pertains to, not about 'other'

File: sympy/geometry/line.py [hunk 15]
```diff
@@ -982,16 +1096,13 @@ def random_point(self, seed=None):
         Returns
         =======
 
-        point : Point
-
-        See Also
-        ========
-
-        sympy.geometry.point.Point
+        Point :
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1099]:
```suggestion
        point : Point
```

File: sympy/geometry/line.py [hunk 16]
```diff
@@ -1024,64 +1154,6 @@ def random_point(self, seed=None):
...
-
-
-        d1 = l1.direction.unit
-        d2 = l2.direction.unit
-
-        bis1 = Line(pt, pt + d1 + d2)
-        bis2 = Line(pt, pt + d1 - d2)
-
-        return [bis1, bis2]
-
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1084]: Why did you remove a whole function? 
Is there a reason for this?

File: sympy/geometry/line.py [hunk 17]
```diff
@@ -1180,7 +1254,7 @@ def __new__(cls, *args, **kwargs):
                 if len(args) > 1:
                     p2 = args[1]
                 else:
-                    p2 = None
+                    p2=None
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1257]: Please keep the spaces around =
For reference check https://www.python.org/dev/peps/pep-0008/

File: sympy/geometry/line.py [hunk 18]
```diff
@@ -1271,7 +1375,9 @@ def equal(self, other):
         return self.equals(other)
 
     def equals(self, other):
-        """Returns True if self and other are the same mathematical entities"""
+        """Returns True if self and other are the same mathematical entities.
+
+        """
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1380]: Keep it as it is

File: sympy/geometry/line.py [hunk 19]
```diff
@@ -1290,8 +1396,9 @@ def plot_interval(self, parameter='t'):
         Returns
         =======
 
-        plot_interval : list (plot interval)
-            [parameter, lower_bound, upper_bound]
+        list :
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1399]:
```suggestion
        plot_interval : list
```

File: sympy/geometry/line.py [hunk 20]
```diff
@@ -1199,12 +1273,24 @@ def __new__(cls, *args, **kwargs):
                 return LinearEntity.__new__(cls, p1, p2, **kwargs)
 
     def contains(self, other):
-        """
-        Return True if `other` is on this Line, or False otherwise.
+        """Checks whether `other` is on Line or not.
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1276]: This shouldn't be about 'other'

File: sympy/geometry/line.py [hunk 21]
```diff
@@ -1383,22 +1490,35 @@ def _svg(self, scale_factor=1., fill_color="#66cc99"):
             Multiplication factor for the SVG stroke-width.  Default is 1.
         fill_color : str, optional
             Hex string for fill color. Default is "#66cc99".
+
         """
+
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1495]: This one is not required

File: sympy/geometry/line.py [hunk 22]
```diff
@@ -1485,7 +1621,9 @@ def distance(self, other):
             return abs(other - self.source)
 
     def equals(self, other):
-        """Returns True if self and other are the same mathematical entities"""
+        """Returns True if self and other are the same mathematical entities.
+
+        """
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1626]: Keep this as it is too.

File: sympy/geometry/line.py [hunk 23]
```diff
@@ -1504,8 +1642,9 @@ def plot_interval(self, parameter='t'):
         Returns
         =======
 
-        plot_interval : list
-            [parameter, lower_bound, upper_bound]
+        list :
```
Comment (Sc0rpi0n101) [inline sympy/geometry/line.py:1645]:
```suggestion
        plot_interval : list
```

File: sympy/geometry/line.py (hunk 1 above)
Comment (mohitshah3111999) [reply sympy/geometry/line.py:31]: Actually, it was showing errors in travis-ci build and there were suggestions to remove this statements. Should I keep this commit or remove?

File: sympy/geometry/entity.py [hunk 24]
```diff
@@ -256,18 +256,11 @@ def bounds(self):
         raise NotImplementedError()
 
     def encloses(self, o):
-        """
-        Return True if o is inside (not on or outside) the boundaries of self.
+        """Return True if o is inside (not on or outside) the boundaries of self.
```
[the remark this answers is not in the document]
Comment (mohitshah3111999) [reply sympy/geometry/entity.py:259]: Can you tell me which section of the documentation I should refer to?
I have gone through DocString formatting section but I didn't find it there.

> 
> 
> This should be left as it is. The text should start from the next line, not the same line.
> See https://docs.sympy.org/dev/documentation-style-guide.html for reference

Can you tell me which section of the documentation I should refer to?
I have gone through DocString formatting section but I didn't find it there.

[the remark this answers is not in the document]
Comment (mohitshah3111999) [reply sympy/geometry/entity.py:259]: > 
> 
> This should be left as it is. The text should start from the next line, not the same line.
> See https://docs.sympy.org/dev/documentation-style-guide.html for reference

Can you tell me which section of the documentation I should refer to?
I have gone through the "DocString formatting" section but I didn't find it there.

File: sympy/geometry/line.py [hunk 25]
```diff
@@ -94,7 +95,10 @@ def _span_test(self, other):
         """Test whether the point `other` lies in the positive span of `self`.
```
[the remark this answers is not in the document]
Comment (mohitshah3111999) [reply sympy/geometry/line.py:93]: I saw in many modules that docstrings are starting from immediately after the triple quotes and in my previous 2 commits, this type of formatting was accepted.
See this : https://github.com/sympy/sympy/pull/19483

File: sympy/geometry/line.py [hunk 26]
```diff
@@ -117,17 +121,20 @@ def ambient_dimension(self):
         Returns
         =======
 
-        dimension : integer
+        int:
```
[the remark this answers is not in the document]
Comment (mohitshah3111999) [reply sympy/geometry/line.py:122]: I think providing a short description of what is returned might be good.
In My last PR https://github.com/sympy/sympy/pull/19483, It was accepted too.

File: sympy/geometry/line.py [hunk 27]
```diff
@@ -117,17 +121,20 @@ def ambient_dimension(self):
...
+            returns the dimension of line.
 
         Examples
         ========
 
+        2-dimensional example.
```
[the remark this answers is not in the document]
Comment (mohitshah3111999) [reply sympy/geometry/line.py:128]: Yes, I will do that.

File: sympy/geometry/line.py [hunk 28]
```diff
@@ -308,17 +341,14 @@ def are_concurrent(*lines):
...
-        True : if the set of linear entities intersect in one point
-        False : otherwise.
-
-        See Also
-        ========
-
-        sympy.geometry.util.intersection
+        bool :
+            True : if the set of linear entities intersect in one point
+            False : otherwise.
```
[the remark this answers is not in the document]
Comment (mohitshah3111999) [reply sympy/geometry/line.py:344]: It's making it the same as others docstrings return statements.
Like,
return type :
    brief explanation.

File: sympy/geometry/line.py (hunk 26 above)
[the remark this answers is not in the document]
Comment (Sc0rpi0n101) [reply sympy/geometry/line.py:122]: I'm not telling you to change the next line. Just this line. 
As I've said later, it should be:
```
entitiy : Type
    Description
```

File: sympy/geometry/entity.py (hunk 24 above)
[the remark this answers is not in the document]
Comment (Sc0rpi0n101) [reply sympy/geometry/entity.py:259]: You can check the example in this section https://docs.sympy.org/dev/documentation-style-guide.html#docstring-guidelines
Or the other examples in the style guide

File: sympy/geometry/line.py (hunk 5 above)
Comment (mohitshah3111999) [reply sympy/geometry/line.py:695]: I actually wanted to show both the answer: true and false
I will change the p4 point to (1, 1, -2).

File: sympy/geometry/line.py (hunk 8 above)
Comment (mohitshah3111999) [reply sympy/geometry/line.py:719]: I think then it's first line summary must be changed as that too will not of any use for user.

File: sympy/geometry/line.py (hunk 14 above)
Comment (mohitshah3111999) [reply sympy/geometry/line.py:1017]: Should I make it like "returns the projection of ``other`` on this linear entity." ?

File: sympy/geometry/line.py (hunk 16 above)
Comment (mohitshah3111999) [reply sympy/geometry/line.py:1084]: I'm sorry.
It was removed by mistake.
Actually, when I opened the pull request for the first time for this file, I was getting a merge conflicts message and then when this function got removed I even don't know.

File: sympy/geometry/line.py (hunk 20 above)
Comment (mohitshah3111999) [reply sympy/geometry/line.py:1274]: I think this is what a function is doing. It's checking whether the point is on it or not.

File: sympy/geometry/line.py (hunk 21 above)
Comment (mohitshah3111999) [reply sympy/geometry/line.py:1493]: I'm sorry, but which one?

File: sympy/geometry/line.py (hunk 23 above)
Comment (mohitshah3111999) [reply sympy/geometry/line.py:1644]: I think I have messed up many things in one commit. I will make changed to this module and will generate a pull request after that.
Thanks for the help.
